### PR TITLE
TSS: nvRead12 expansion

### PR DESCRIPTION
**Pull request by ChriMarMe**
-dependency update
-nvread12 reads with noAuth/OwnerAuth/BlobAuth and calls the correct function from go-tpm

### CODE REVIEW

File: pkg/tss/nvram.go [hunk 1]
```diff
@@ -14,12 +15,34 @@ import (
...
+	if indexData == nil {
+		return nil, fmt.Errorf("index not found")
+	}
+
+	// Check if authData is needed
+	// AuthRead 0x00200000 | OwnerRead 0x00100000
+	needAuthData := indexData.Permission.Attributes & (0x00100000 | 0x00200000)
+
+	// Check for AuthRead
+	authread := indexData.Permission.Attributes & 0x00200000
```
Comment (ChriMarMe) [inline pkg/tss/nvram.go:34]: This is not correct yet

File: pkg/tss/nvram.go [hunk 2]
```diff
@@ -14,12 +15,34 @@ import (
...
+
+	if needAuthData == 1 {
+		if authread == 0 {
```
Comment (ChriMarMe) [inline pkg/tss/nvram.go:37]: This must be 0 if the upper review is implemented

File: pkg/tss/nvram.go [hunk 3]
```diff
@@ -6,20 +6,41 @@ package tss
...
+		return nil, err
 	}
-	return tpm1.NVReadValue(rwc, index, offset, len, []byte(ownAuth[:20]))
+	if indexData == nil {
+		return nil, fmt.Errorf("index not found")
+	}
+
+	// Check if authData is needed
+	// AuthRead 0x00200000 | OwnerRead 0x00100000
+	needAuthData := 1 >> (indexData.Permission.Attributes ^ (0x00200000 | 0x00100000))
```
Comment (hugelgupf) [inline pkg/tss/nvram.go:31]: what are these magic constants? can you name them?

Comment (ChriMarMe) [reply pkg/tss/nvram.go:31]: oh. sure :)

**Pull request merged**